fish-rust/src/common.rs

```diff
@@ -318,57 +318,144 @@ fn escape_string_script(input: &wstr, flags: EscapeFlags) -> WString {
     out
 }
 
-/// Escape a string in a fashion suitable for using as a URL. Store the result in out_str.
-#[widestrs]
+/// Escape a string in a fashion suitable for using as a URL.
+///
+/// Url-safe ASCII input chars are passed through while everything else is UTF-8 percent-encoded
+/// (converted to UTF-8 then each byte encoded as a hex nibble preceded by a %).
 fn escape_string_url(input: &wstr) -> WString {
-    let narrow = wcs2string(input);
-    let mut out = WString::new();
-    for byte in narrow.into_iter() {
-        if (byte & 0x80) == 0 {
-            let c = char::from_u32(u32::from(byte)).unwrap();
-            if c.is_alphanumeric() || [b'/', b'.', b'~', b'-', b'_'].contains(&byte) {
```

Comment:
The first equivalent of "is ascii" check is not needed since we can check directly for the chars we want.